File: backend/F1Fantasy/Models/Group.cs
namespace F1Fantasy.Models;

public class Group
{
    public int Id { get; set; }
    public required string Name { get; set; }
    public required string InviteCode { get; set; }
    public required string LockMode { get; set; } // "admin", "system", "hybrid"
    public required string AdminUserId { get; set; } // Clerk user ID
    public DateTime CreatedAt { get; set; }
    
    // Prediction lock (group-wide for all predictions)
    public bool PredictionsLocked { get; set; }
    public DateTime? LockedAt { get; set; }

    // Navigation properties
    public ICollection<GroupMember> Members { get; set; } = new List<GroupMember>();
}

File: backendJava/f1fantasy/src/main/java/no/f1fantasy/entity/Group.java
package no.f1fantasy.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "Groups",
        indexes = {
                @Index(name = "IX_Groups_InviteCode", columnList = "InviteCode", unique = true),
                @Index(name = "IX_Groups_AdminUserId", columnList = "AdminUserId")
        })
@Data
public class Group {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "Id")
    private Integer id;

    @Column(name = "Name", length = 200, nullable = false)
    private String name;

    @Column(name = "InviteCode", length = 50, nullable = false)
    private String inviteCode;

    /** Values: "admin", "system", "hybrid" */
    @Column(name = "LockMode", length = 20, nullable = false)
    private String lockMode;

    @Column(name = "AdminUserId", length = 100, nullable = false)
    private String adminUserId;

    @Column(name = "CreatedAt", nullable = false)
    private OffsetDateTime createdAt;

    @Column(name = "PredictionsLocked", nullable = false)
    private boolean predictionsLocked;

    @Column(name = "LockedAt")
    private OffsetDateTime lockedAt;

    @OneToMany(mappedBy = "group", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<GroupMember> members = new ArrayList<>();
}
